robot-name/ena/robot_name.go

```diff
@@ -0,0 +1,57 @@
+package robotname
+
+import (
+	"errors"
+	"math"
+	"math/rand"
+	"time"
+)
+
+var usedNames = make(map[string]bool)
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+type Robot struct {
+	name string
```

Comment:
I would export the name or hide the struct itself.

Reply:
Why? It's like class with private fields managed by getter/setter methods.

Reply:
Actually scratch my comment. I don't know what I was thinking about. You have a Name function.